### PR TITLE
[backport] Updating the updater

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
Backport from Chef-18. This updates how we manage version changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .expeditor/update_version.sh [hunk 1]
```diff
@@ -12,6 +12,7 @@
 set -evx
 
 VERSION=$(cat VERSION)
+ORIGINAL_VERSION=$(git show main:VERSION)
```
Comment (neha-p6) [inline .expeditor/update_version.sh:15]: This needs to be changed for chef 17
`git show chef-17:VERSION`

**Review by neha-p6 — CHANGES_REQUESTED** [review]

File: .expeditor/update_version.sh (hunk 1 above)
Comment (johnmccrae) [reply .expeditor/update_version.sh:15]: oh crap, I cna't believe I missed that thanks

Comment (johnmccrae) [reply .expeditor/update_version.sh]: fixed

**Review by neha-p6 — APPROVED** [review]

**Comment by neha-p6** [comment]
This pr backports https://github.com/chef/chef/pull/13005

**Review by stringsn88keys — APPROVED** [review]

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**